nix/patches/nixpkgs-revert-ghc-bootstrap.patch

```diff
@@ -52,29 +53,29 @@ index 41af279e83f..5853a470efc 100644
    }.${stdenv.hostPlatform.system}
      or (throw "cannot bootstrap GHC on this platform"));
 diff --git a/pkgs/top-level/haskell-packages.nix b/pkgs/top-level/haskell-packages.nix
-index 3e85109527e..388210b83fb 100644
+index 693f8fe8276..034294b7264 100644
 --- a/pkgs/top-level/haskell-packages.nix
 +++ b/pkgs/top-level/haskell-packages.nix
-@@ -4,7 +4,8 @@ let
+@@ -4,8 +4,8 @@ let
    # These are attributes in compiler and packages that don't support integer-simple.
    integerSimpleExcludes = [
      "ghc822Binary"
 -    "ghc865Binary"
+     "ghc8102Binary"
 +    "ghc863Binary"
-+    "ghc844"
      "ghcjs"
      "ghcjs86"
      "integer-simple"
-@@ -47,7 +48,7 @@ in {
-
+@@ -48,7 +48,7 @@ in {
+ 
      ghc822Binary = callPackage ../development/compilers/ghc/8.2.2-binary.nix { };
-
+ 
 -    ghc865Binary = callPackage ../development/compilers/ghc/8.6.5-binary.nix { };
 +    ghc863Binary = callPackage ../development/compilers/ghc/8.6.3-binary.nix { };
-
-     ghc865 = callPackage ../development/compilers/ghc/8.6.5.nix {
-       bootPkgs = packages.ghc822Binary;
-@@ -56,31 +57,25 @@ in {
+ 
+     ghc8102Binary = callPackage ../development/compilers/ghc/8.10.2-binary.nix {
```

Comment:
Currently we're using `ghc883`:
https://github.com/PostgREST/postgrest/blob/d395bb60522aa228a534d9181339519f171fe6b2/default.nix#L5-L6

Can I switch that to `ghc8102Binary` to use GHC 8.10 and expect to all work the same?
GHC 8.10 might contain some memory usage improvements and I was thinking to benchmark it :)

Reply:
Oh, actually I've just tried that and I get a nix failure:
```
error: attribute 'haskellCompilerName' missing, at /nix/store/yacxa1kgxalkyw7fyk27xm76vp7gjqwd-source/pkgs/development/haskell-modules/make-package-set.nix:136:32
```

No problem though. That was more of a question.

Reply:
@steve-chavez it probably works for normal builds, but for the fully static build we need `static-haskell-nix` to catch up: https://github.com/nh2/static-haskell-nix/pull/98